automation/src/main/java/org/wso2/testgrid/automation/parser/FunctionalTestResultParser.java

```diff
@@ -0,0 +1,154 @@
+                        String elementName = startElement.getName().getLocalPart();
+                        if (HTTP_SAMPLE_ELEMENT.equalsIgnoreCase(elementName) ||
+                                SAMPLE_ELEMENT.equalsIgnoreCase(elementName)) {
+                            testCase = this.addAttributes(startElement);
+                        } else if (FAILURE_MESSAGE_ELEMENT.equalsIgnoreCase(elementName)) {
+                            failureMsgElement = true;
+                        }
+                        break;
+                    case XMLStreamConstants.CHARACTERS:
+                        Characters characters = event.asCharacters();
+                        if (failureMsgElement) {
+                            testCase.setFailureMessage(characters.getData());
+                            failureMsgElement = false;
+                        }
+                        break;
+                    case XMLStreamConstants.END_ELEMENT:
+                        EndElement endElement = event.asEndElement();
+                        String nodeName = endElement.getName().getLocalPart();
+                        if (HTTP_SAMPLE_ELEMENT.equalsIgnoreCase(nodeName) ||
+                                SAMPLE_ELEMENT.equalsIgnoreCase(nodeName)) {
+                            this.testScenario.addTestCase(testCase);
+                        }
+                        break;
+                }
+            }
+
+            if (logger.isDebugEnabled()) {
+                logger.debug("End parsing scenario-results file of the TestScenario : '" + testScenarioName +
+                        "' using the FunctionalTestResultParser");
+            }
+        } catch (XMLStreamException e) {
+            throw new JMeterResultParserException("Unable to parse the scenario-results file of TestScenario :" +
+                    testScenarioName, e);
+        } catch (FileNotFoundException e) {
+            throw new JMeterResultParserException("Unable to locate the scenario-results file.", e);
+        } finally {
+            try {
+                if (inputStream != null) {
+                    inputStream.close();
+                }
+            } catch (IOException e) {
+                throw new JMeterResultParserException("Unable to close the input stream of scenario results file of " +
+                        "the TestScenario : " + testScenarioName, e);
+            }
+        }
+    }
+
+    private TestCase addAttributes(StartElement startElement) {
+        TestCase testCase = new TestCase();
+        testCase.setTestScenario(this.testScenario);
+        Iterator<Attribute> attributes = startElement.getAttributes();
+
+        while (attributes.hasNext()) {
+            Attribute attribute = attributes.next();
+            if (TEST_NAME_ATTRIBUTE.equals(attribute.getName().getLocalPart())) {
+                testCase.setName(attribute.getValue());
+            } else if (TEST_STATUS_ATTRIBUTE.equals(attribute.getName().getLocalPart())) {
```

Comment:
We could set the TestCase as successful if the "assertionResult" node does not contain any error or failure message. But the attribute "ec" was there in every test case in identity-integration tests.

Reply:
OK, if you think assertionResult node is a better estimate, then we should use that.
May be the 'ec' attribute was there for identity jmeter scripts because of some convention they follow. It's not fool-proof as I've seen.